src/Umbraco.Infrastructure/PublishedContentQuery.cs

```diff
@@ -93,14 +93,14 @@ private static bool ConvertIdObjectToUdi(object id, out Udi guidId)
 
         #region Content
 
-        public IPublishedContent Content(int id) => ItemById(id, _publishedSnapshot.Content);
+        public IPublishedContent Content(int id) => ItemById(id, _publishedSnapshotAccessor.PublishedSnapshot.Content);
```

Comment:
There might not be a published snapshot, these need null checks and then to throw a meaningful exception explaining why. It won't work without a snapshot which requires a current umbraco context

Reply:
Alternatively we could add `IUmbracoHelperAccessor` and `IPublishedContentQueryAccessor`? It will be a fairly used usecase to need one of these from a singleton. 

I know this will just be syntactical sugar for injecting IHttpContextAccessor and use that as ServiceProvider.

Reply:
I think that would make things a lot clearer as to what to expect. I just don't see how people are going to use these in singletons though. The only way that will be possible is that within the singleton they use the UmbracoContextFactory and while using the resulting context then use one of these accessors.